os/kernel/log_dump/log_dump.c

```diff
@@ -505,6 +505,26 @@ int log_dump_save(char ch)
 	return LOG_DUMP_OK;
 }
 
+size_t log_dump_get_size(void)
+{
+	size_t size = 0;
+	struct log_dump_chunk_s* head = (struct log_dump_chunk_s *)sq_peek(&log_dump_chunks);
+	while (head != (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks)) {
+		head = (struct log_dump_chunk_s *)sq_next((sq_entry_t *)head);
+		size += CONFIG_LOG_DUMP_CHUNK_SIZE;
+	}
+	/* compress_curbytes is the length of compressed logs in latest log_dump_chunk */
+	size += compress_curbytes;
+
+	/* This is the size of compressed partially filled uncomp_buf.
+	 * Note: It is recommended to use this log_dump_get_size api after
+	 * log_dump_read_wake() is performed, so that it has correct value of last_comp_block_size.
+	 * Else it might add old last_comp_block_size which is already present in log_dump_chunk list */
+	size += last_comp_block_size;
```

Comment:
You return compressed log size + uncompressed log size. When we read saved log dump, don't we compress uncompressed log?? If we don't, then how can we know how many data should be uncompressed and how many data should be used as it is?

Reply:
Earlier, when we were opening the procfs file of log dump, it was compressing the partially filled log buffer. Now the compressing the partially filled log buffer has been shifted to when log dump save is stopped . So that whenever we stop log dump process, we will compress the partially filled log buffer and update the size. 
two cases while getting log dump size:
1. When log dump save is true and log dump is saving character : In this case , log dump size will return the size of compressed logs till now in the log_dump_chuck and it will not include last_comp_block_size becuase it is not valid.
2. when log dump save is false and log dump is not saving : In this case , log dump size will include compressed logs in log_dump_chunk + last_comp_block_size becuase after log dump save is stopped we will compress the last partially filled logs.